WoWsShipBuilder.Web/Dialogs/HideComparisonColumnsDialog.razor

```diff
@@ -0,0 +1,79 @@
+﻿@using WoWsShipBuilder.ViewModels.Other
+@inject ILocalizer Localizer
+
+<MudDialog Style="max-height: 95vh" DefaultFocus="DefaultFocus.Element">
+    <DialogContent>
+        <div style="max-height: 85vh; overflow: scroll;">
+            <MudText Typo="Typo.body2" Align="Align.Center" Style="font-weight: bold" Class="mb-1">@Localizer.SimpleAppLocalization(nameof(Translation.ShipComparison_ActiveColumns)).ToUpper()</MudText>
+            <MudStack Row>
+                @foreach (var (section, columns) in structure)
+                {
+                    <MudPaper Outlined Elevation="0" Square="true" @key="@(section.ShipComparisonDataSectionToString())">
+                        <MudStack Justify="Justify.Center" AlignItems="AlignItems.Center">
+                            <MudText Typo="Typo.body2" Align="Align.Center" Style="font-weight: bold">
+                                @(section.LocalizeShipComparisonDataSection(Localizer).ToUpper())
+                            </MudText>
+                            @foreach (var column in columns)
+                            {
+                                <MudSwitch @bind-Checked="@(structure[section][column.Key])" Label="@column.Key" LabelPosition="LabelPosition.End" Size="Size.Medium" Color="Color.Warning" @key="@($"{section}_{column}")" Style="width: 100%;" Class="ml-1"/>
+                            }
+                        </MudStack>
+                    </MudPaper>
+                }
+            </MudStack>
+        </div>
+    </DialogContent>
+    <DialogActions>
+        <MudButton OnClick="Cancel" Variant="Variant.Text">@Localizer.GetAppLocalization(nameof(Translation.Cancel)).Localization</MudButton>
+        <MudButton OnClick="Confirm" Variant="Variant.Text">@Localizer.GetAppLocalization(nameof(Translation.Dialog_Save)).Localization</MudButton>
+    </DialogActions>
+</MudDialog>
+
+@code
+{
+    [CascadingParameter]
+    MudDialogInstance MudDialog { get; set; } = new();
+
+    [Parameter, EditorRequired]
+    public Dictionary<string, bool> HiddenColumns { get; set; } = new();
+
+    private readonly Dictionary<ShipComparisonDataSections, Dictionary<string, bool>> structure = new();
+
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+
+        foreach (string key in HiddenColumns.Keys)
+        {
+            string[] split = key.Split('_');
+            var section = Enum.Parse<ShipComparisonDataSections>(split[0]);
+            if (structure.TryGetValue(section, out Dictionary<string, bool>? dataSection))
```

Comment:
Can this ever be true? It looks like you only add the values from the dictionary to the structure in this method (OnInitialized is the first method called in this component) so this should never evaluate to true anyways

Reply:
there are multiple keys with the same section. after the first key of the section all the others will return true untile the section changes

Reply:
there are multiple keys within the same section. after the first key of the section is added all the others will return true until the section changes

Reply:
How can there be multiple keys for the same section? That already implies that something went wrong. Keys should be unique, if they are not, that implies an issue with the selection of keys